transcoders.go

```diff
@@ -238,13 +238,66 @@ func garlic64BtS(b []byte) (string, error) {
 	return addr, nil
 }
 
-func garlicValidate(b []byte) error {
+func garlic64Validate(b []byte) error {
 	if len(b) < 386 {
 		return fmt.Errorf("failed to validate garlic addr: %s not an i2p base64 address. len: %d\n", b, len(b))
 	}
 	return nil
 }
 
+var TranscoderGarlic32 = NewTranscoderFromFunctions(garlic32StB, garlic32BtS, garlic32Validate)
+
+var garlicBase32Encoding = base32.NewEncoding("abcdefghijklmnopqrstuvwxyz234567")
+
+func garlic32StB(s string) ([]byte, error) {
+	s = strings.Replace(s, ".b32.i2p", "", -1)
+	// garlic address without the ".b32.i2p" substring
+	if len(s) < 55 {
+		if len(s) != 52 {
+			return nil, fmt.Errorf("failed to parse garlic addr: %s not a i2p base32 address. len: %d", s, len(s))
+		}
+	}
+
+	padout := func(s2 string) string {
```

Comment:
Defining this as a function seems unnecessary.

Reply:
It was wrong anyway, thanks! Fixed in the new changes.

Reply:
Something seems to be off with my math. Just a moment.

Reply:
The reason why was that the padding required is could be shorter, longer, or 0 based, with 4 for regular .b32.i2p addresses, zero for 56 char encrypted leaseset .b32.i2p, or variable non-zero lengths for addresses with secrets. But then I forgot to do that.